ui/lib/core/addon/components/download-button.js

```diff
@@ -1,51 +1,67 @@
-import Component from '@ember/component';
-import { computed } from '@ember/object';
-import hbs from 'htmlbars-inline-precompile';
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+/**
+ * @module DownloadButton
+ * DownloadButton components are an action button used to download data. Both the action text and icon are yielded.
+ *
+ * @example
+ * ```js
+ *   <DownloadButton
+ *     @buttonClass="toolbar-link"
+ *     @data={{this.data}}
+ *     @filename={{this.filename}}
+ *     @mime={{this.mime}}
+ *     @extension={{this.extension}}
+ *     @stringify={{true}}
+ *   >
+ *    <Icon @name="download" />
+ *      Download
+ *   </DownloadButton>
+ * ```
+ * @param {string} data - data to download
+ * @param {boolean} [stringify=false] - argument to stringify the data before passing to the File constructor
+ * @param {string} [filename] - name of file that prefixes the ISO timestamp generated when download
+ * @param {string} [mime='text/plain'] - media type to be downloaded
+ * @param {string} [extension='txt'] - file extension
+ * @param {string} [buttonClass] - override css class defaults to customize how the button looks
+ */
 
-export default Component.extend({
-  layout: hbs`{{#if (has-block)}} {{yield}} {{else}} {{actionText}} {{/if}}`,
-  tagName: 'a',
-  role: 'button',
-  attributeBindings: ['role', 'download', 'href'],
-  download: computed('filename', 'extension', function () {
-    return `${this.filename}-${new Date().toISOString()}.${this.extension}`;
-  }),
+export default class DownloadButton extends Component {
+  get extension() {
+    return this.args.extension || 'txt';
+  }
 
-  fileLike: computed('data', 'mime', 'stringify', 'download', function () {
-    let file;
-    let data = this.data;
-    const filename = this.download;
-    const mime = this.mime;
-    if (this.stringify) {
-      data = JSON.stringify(data, null, 2);
-    }
-    if (window.navigator.msSaveOrOpenBlob) {
-      file = new Blob([data], { type: mime });
-      file.name = filename;
```

Comment:
Oh! I meant to make a comment - thank you. This was added by #4376 because IE and Edge didn't like the File constructor. Since [2020](https://caniuse.com/?search=File) it's been supported! Going to add this to the PR description for posterity.